services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/client/CloudDccFeignHttpClientProvider.java

```diff
@@ -48,7 +61,24 @@ public Client createFeignClient() {
   private ApacheHttpClientFactory federationHttpClientFactory() {
     return new DefaultApacheHttpClientFactory(HttpClientBuilder.create()
         .setMaxConnPerRoute(connectionPoolSize)
-        .setMaxConnTotal(connectionPoolSize));
+        .setMaxConnTotal(connectionPoolSize)
+        .setSSLContext(getSslContext(this.trustStore, this.trustStorePassword)));
+  }
+
+  private SSLContext getSslContext(File trustStorePath, String trustStorePass) {
+    logger.info("Instantiating SSL context with keystore: " + trustStorePath.getName());
+    try {
+      return SSLContextBuilder.create().loadTrustMaterial(trustStorePath,
+              emptyCharrArrayIfNull(trustStorePass))
+          .build();
+    } catch (Exception e) {
+      logger.error("Problem on creating SSL context with key: " + trustStorePath.getName(), e);
```

Comment:
I would change this a bit, as "key" might be misleading here... how about
"Problem creating SSL context with truststore "...